tidy3d/components/geometry/utils_2d.py

```diff
@@ -63,17 +59,37 @@ def set_bounds(geom: Geometry, bounds: Tuple[float, float], axis: Axis) -> Geome
         new_center[axis] = (bounds[0] + bounds[1]) / 2
         new_length = bounds[1] - bounds[0]
         return geom.updated_copy(center=new_center, length=new_length)
+    if isinstance(geom, Transformed):
```

Comment:
when I see these chains of `if isinstance()`, it tends to suggest to me that these should be written as methods of the class, and then just called once, passing the instance in with whatever arguments are needed. for example, would it make sense for something like a `Geometry.update_from_bounds(self, bounds: Bounds) -> Geometry` method implemented in each of these classes? In this example, perhaps we could just replace all of this code with `geom.update_from_bounds(new_bounds)`?

Reply:
Yea I agree! One complication is that this change is from a commit introduced in a [different PR](https://github.com/flexcompute/tidy3d/pull/1644). I need to use the changes in that PR or these coaxial ports are not possible. So right now, I have rebased onto that branch.

I'll address this in the other PR, unless you have a different suggestion.

Reply:
ah I see, I think what you suggest works.